data_models.py

```diff
@@ -288,6 +322,20 @@ def compile_reconstruction(self, test_set=False):
                                                   index=input_df.index,
                                                   columns=input_df.columns)
 
+        if hasattr(self, 'ica_df'):
```

Comment:
Same thing as above. I'm not hugely familiar with these objects, but it seems like you could make a generalizable function and not have to redo the code every time you introduce a new method.